presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/MySqlShardDao.java

```diff
@@ -39,4 +40,14 @@
     @SqlBatch("INSERT IGNORE INTO deleted_shards (shard_uuid, delete_time)\n" +
             "VALUES (:shardUuid, CURRENT_TIMESTAMP)")
     void insertDeletedShards(@Bind("shardUuid") Iterable<UUID> shardUuids);
+
+    @SqlUpdate("DELETE FROM transactions\n" +
+            "WHERE end_time < :maxEndTime AND (\n" +
```

Comment:
We can try this out on a large database and see if there is any important difference in performance.